src/is-of-type.ts

```diff
@@ -0,0 +1,68 @@
+import { Action, AnyAction } from './create-action'
+import { ActionCreator } from './create-action-creator'
+import { getType } from './get-type'
+
+export function isOfType<
+  TType extends string | AnyAction | ActionCreator<string>,
+  TAction extends AnyAction
+>(
+  type: TType | TType[],
+  action: TAction
+): action is TAction extends Action<
+  TType extends string
+    ? TType
+    : TType extends AnyAction | ActionCreator<string>
+    ? TType['type']
+    : never
+>
+  ? TAction
+  : never
+
+export function isOfType<
+  TType extends string | AnyAction | ActionCreator<string>
+>(
+  type: TType | TType[]
+): <TAction extends AnyAction>(
+  action: TAction
+) => action is TAction extends Action<
+  TType extends string
+    ? TType
+    : TType extends AnyAction | ActionCreator<string>
+    ? TType['type']
+    : never
+>
+  ? TAction
+  : never
+
+export function isOfType<
+  TType extends string | AnyAction | ActionCreator<string>,
+  TAction extends AnyAction
+>(typeOrTypes: TType | TType[], action?: TAction) {
+  const types = Array.isArray(typeOrTypes) ? typeOrTypes : [typeOrTypes]
```

Comment:
You can use `castArray` function from utils.

Reply:
completed!